gbench/clblast/blas1/asum.cpp

```diff
@@ -0,0 +1,93 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) 2016 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename asum.cpp
+ *
+ **************************************************************************/
+
+#include "utils.hpp"
+
+
+template <typename scalar_t>
+void BM_Asum(benchmark::State& state) {
+  // Standard test setup.
+  const index_t size = static_cast<index_t>(state.range(0));
+  double size_d = static_cast<double>(size);
+  state.counters["size"] = size_d;
+  state.counters["n_fl_ops"] = 2 * size_d;
+  state.counters["bytes_processed"] = size_d * sizeof(scalar_t);
+
+  // Create data
+  std::vector<scalar_t> v1 = benchmark::utils::random_data<scalar_t>(size);
+  scalar_t vr;
+
+  ExecutorType* ex = getExecutor().get();
```

Comment:
rename this function to getTheExecutor()

Reply:
Done